app/models/concerns/profile_searching.rb

```diff
@@ -38,6 +41,19 @@ module ProfileSearching
       test_results = TestResult.select(:profile_id).distinct
       has_test_results && where(id: test_results) || where.not(id: test_results)
     }
+    scope :has_policy_test_results, lambda { |has_test_results = true|
+      # the use of default scope is to bypass Pundit
+      # and avoid the situation with bind parameters
+      # that must have a different order in the resulting query
+      # of a scoped_search
+      policy_test_results = default_scoped.joins(policy_object: :test_results)
```

Comment:
Nit: this variable makes it look like a collection of test results instead of a collection of profiles. `with_policy_test_results` or similar would be more appropriate.

Reply:
It makes the line longer. :-)

Reply:
Very astute! ;) Readability is far more important to me than comments, and variable naming plays a huge role in that. `policy_test_results` may well become a relationship on profile one day (i.e. `has_many :policy_test_results, through: :policy_object, source: :test_results`).

Reply:
Changed.